### DISCUSSION TITLE
Vercel seems to be stripping my credentials in next-auth request? (app folder)

### DISCUSSION THREAD

**Discussion by JinMABrett** [vercel/community#3041] [Help] [answered]
### Summary

So i've been having an issue with next-auth using the app router. Everything works just fine on localhost but when i push to Vercel, my API request seems to contain empty request body. Credentials are logging the request just fine before fetch and in the network tab, but i keep getting this error on my api call

```vbnet
- error SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:6571:19)
    at successSteps (node:internal/deps/undici/undici:6545:27)
    at node:internal/deps/undici/undici:1211:60
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  ```

I'm a new coder and have spent the last 2 days exhausting every idea i could come up with, any help would be appreciated. I'm thinking it could be a vercel issue since it only happens when deployed but works fine locally. I've double and tripple checked environment variables and they are setup just fine on vercel. 

For much more context here is my next-auth discussion with all my relevant code (or at least what I think is relevant code)

https://github.com/nextauthjs/next-auth/discussions/7887

### Steps to Reproduce

Next.js 13.4 project with app directory
setup next-auth
push to Vercel
Get error.

**Comment by Callumk7** [comment]
I am having the same problem, where the body of my request seems to be missing in the vercel function but not on localhost or other providers.

**Comment by jamesallchin-rb** [comment]
Same for me

**Comment by JinMABrett** [comment, marked as answer]
For anybody interested, [jamesallchin-rb](https://github.com/jamesallchin-rb) was able to solve this issue for me here

https://github.com/nextauthjs/next-auth/discussions/7887#discussioncomment-6646858

He was able to solve this issue by using axios rather than fetch in the next-auth credentials config.
